### PR TITLE
feat(Modal): add description property

### LINKED ISSUE #3777: PF4 Modal: Modal enhancements for table column management
**Describe the issue. What is the expected and unexpected behavior?**
related core issue https://github.com/patternfly/patternfly-next/issues/2636

### PR DESCRIPTION
**Pull request by kmcfaul**
+ move ID to description when it exists.

**What**: Closes #3777

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/Modal/ModalContent.tsx [hunk 1]
```diff
@@ -29,6 +30,8 @@ export interface ModalContentProps {
   isOpen?: boolean;
   /** Complex header (more than just text), supersedes title for header content */
   header?: React.ReactNode;
+  /** Description */
+  description?: React.ReactNode;
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Modal/ModalContent.tsx]: SHouldn't we also add this prop to `Modal` so the consumer knows that is where they pass it?

Comment (mcoker) [inline packages/patternfly-4/react-core/src/components/Modal/ModalContent.tsx]: Does this mean the description can contain markup, too, and doesn't have to just be a string?

Comment (tlabaj) [reply packages/patternfly-4/react-core/src/components/Modal/ModalContent.tsx]: @mcoker yes.

File: packages/patternfly-4/react-core/src/components/Modal/examples/Modal.md [hunk 2]
```diff
@@ -62,6 +64,58 @@ class SimpleModal extends React.Component {
...
+    return (
+      <React.Fragment>
+        <Button variant="primary" onClick={this.handleModalToggle}>
+          Show Modal
+        </Button>
+        <Modal
+          title="Modal Header"
+          isOpen={isModalOpen}
+          onClose={this.handleModalToggle}
+          description="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/Modal/examples/Modal.md]: Can you change the description to match core so it's clear which text is the description? The text from core is:
> A description is used when you want to provide more info about the modal than the title is able to describe. The content in the description is static and will not scroll with the rest of the modal body.

File: packages/patternfly-4/react-core/src/components/Modal/examples/Modal.md [hunk 3]
```diff
@@ -62,6 +64,58 @@ class SimpleModal extends React.Component {
 }
 ```
 
+```js title=Basic-with-description
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/Modal/examples/Modal.md]: Change to just "With description" to be consistent with https://pf4.patternfly.org/documentation/core/components/modalbox#with-description

**Pull request merged**